### PR TITLE
Added feedback for failed SIS login, simplified logged in display

### PR DESCRIPTION
**Pull request by hannesmcman**
I started this PR with the intention of resolving #1907...I realized halfway through that this was already resolved when I checked the live version on my phone, but it wasn't working on master so I stuck it out. I modified the failed login feedback to be an alert and added a similar alert if the user is not connected to the internet. I also simplified the display of the login-credentials section on the settings view so that the inputs disappear when the user is logged in. I think the showNetworkFailureMessage() can be moved to another module and imported, but I wasn't sure where to put it so I thought I'd ask.

### CODE REVIEW

File: source/flux/parts/settings.js
Comment (hawkrives) [inline source/flux/parts/settings.js:223]: Omg. Whoops. Good catch!

File: source/flux/parts/settings.js
Comment (hawkrives) [inline source/flux/parts/settings.js]: I don't mind having these in this file; however, given that we only call them once, what about inlining them into the logInViCredentials function itself?

**Review by hawkrives — COMMENTED** [review]
In general, wow! Thanks. 

I want to think about the wording of the alerts before I approve the PR, but that's all I'm holding off on. 

Thanks again!

File: source/flux/parts/settings.js
Comment (hannesmcman) [reply source/flux/parts/settings.js]: Is this where they should be? Or should they be above the return function? https://github.com/StoDevX/AAO-React-Native/pull/2019/commits/d1f2b9d71a733dc4d391d66d51dbbee20e4e6e4f

File: source/flux/parts/settings.js
Comment (hawkrives) [inline source/flux/parts/settings.js]: Ah. I was thinking just inline the `Alert.alert` calls directly into the spots they're called from:

```js
  if (!isConnected) {
    dispatch({ type: CREDENTIALS_LOGIN_FAILURE });
    Alert.alert(
      "Network Failure",
      "You are not connected to the internet. Please connect if you want to access this feature.",
      [{ text: "OK" }]
    );
  }
  const result = await performLogin(username, password);
  if (result) {
    dispatch({
      type: CREDENTIALS_LOGIN_SUCCESS,
      payload: { username, password }
    });
    // since we logged in successfully, go ahead and fetch the meal info
    dispatch(updateBalances());
  } else {
    dispatch({ type: CREDENTIALS_LOGIN_FAILURE });
    Alert.alert(
      "Invalid Login",
      "The username and password you provided do not match a valid account. Please try again.",
      [{ text: "OK" }]
    );
  }
```

But, I'm OK with having the shorter names to refer to them by.

---

Thus, I'll change my request:

Would you please move these two function definitions back to the top level, and define them above `logInViaCredentials`, please? 

And could you also change them to arrow functions, please? For example, 

```js
const showNetworkFailureMessage = () =>
  Alert.alert(
    "Network Failure",
    "You are not connected to the internet. Please connect if you want to access this feature.",
    [{ text: "OK" }]
  )
```

The extra line taken up by the closing brace just seems wasteful to me, since the function's only one statement long 😜

**Comment by hannesmcman** [comment]
https://github.com/StoDevX/AAO-React-Native/pull/2019/commits/276fe370221fa3df8bafd9be0be4a4745b48edc8 resolves an issue where the Network Error alert would fire after an internet connection was established after an attempt to login while offline.

File: source/flux/parts/settings.js
Comment (hawkrives) [inline source/flux/parts/settings.js]: Is this import used?

File: source/lib/login.js
Comment (hawkrives) [inline source/lib/login.js]: Is this import used?

File: source/flux/parts/settings.js
Comment (hawkrives) [inline source/flux/parts/settings.js]: I'd like to leave this here, just to short-circuit the logic when we know we're offline.

File: source/lib/login.js
Comment (hawkrives) [inline source/lib/login.js]: Could you change this to something like 

```js
    const form = buildFormData({username, password})
    
    let loginResult = null
    try {
      loginResult = await fetch(OLECARD_AUTH_URL, {
        method: 'POST',
        body: form,
      })
    } catch (err) {
      return false
    }

    if (!loginResult) {
      return false
    }

    const page = await loginResult.text()

    if (page.includes('Password')) {
```

I don't like to assign to things outside of a Promise from a promise callback … it feels fragile. Instead, I used a normal try/catch block around the `await` keyword to catch the network error, and return false from the catch block.

**Review by hawkrives — CHANGES_REQUESTED** [review]

File: source/flux/parts/settings.js
Comment (hannesmcman) [reply source/flux/parts/settings.js]: Isn't this redundant with the if else clause below? If I put this back, it would trigger two alerts if the user wasn't connected to the internet..

Comment (hannesmcman) [reply source/flux/parts/settings.js]: The reason I replaced this if statement with a combination of the try-catch in login.js and 
```javascript
else {
      dispatch({type: CREDENTIALS_LOGIN_FAILURE})
      if (isConnected) {
        showInvalidLoginMessage()
      } else {
        showNetworkFailureMessage()
      }
}
``` 
in settings.js was that sometimes the redux store wouldn't update the app.isConnected variable correctly...thus, even if the login was successful, the "network error" alert would still be shown. This way, the alert will only go off if both the login is unsuccessful and app.isConnected is false.

**Comment by hawkrives** [comment]
You'll probably need to run `yarn prettier` on this branch. I'm going to make a quick PR to re-enable Danger so it reminds us about these things.

**Review by hawkrives — APPROVED** [review]
Approving pending re-prettification 😸

**Review by drewvolz — APPROVED** [review]
Great work, @hannesmcman

**Comment by hawkrives** [comment]
:party:

**Pull request merged**
